pkg/hpa/service.go

```diff
@@ -72,6 +74,7 @@ func New(
 		tortoiseHPATargetUtilizationMaxIncrease: tortoiseHPATargetUtilizationMaxIncrease,
 		recorder:                                recorder,
 		tortoiseHPATargetUtilizationUpdateInterval: tortoiseHPATargetUtilizationUpdateInterval,
+		defaultHPABehavior:                         defaultHPABehavior,
```

Comment:
You can put `defaultHPABehavior` (defined at L224) if `defaultHPABehavior` (at L55) is nil. Then, we can remove `getDefaultHPABehavior` function because `Service.defaultHPABehavior` would be always non-nil